Project/GNU/CLI/test/helpers.sh

```diff
@@ -1,17 +1,221 @@
+
 if type -p md5sum ; then
     md5cmd="md5sum"
 elif type -p md5 ; then
     md5cmd="md5 -r"
 else
-    exit 1
+    fatal "internal" "command not found for md5sum" >&${fd}
 fi >/dev/null 2>&1
 
+case "${OSTYPE}" in
+    linux*)
+        fsize="stat -c %s"
+        ;;
+    darwin*|bsd*)
+        fsize="stat -f %z"
+        ;;
+  *)
+        fatal "internal" "unsupported platform"
+        ;;
+esac
+
 fd=1
-if command exec >&9 ; then
+if (command exec >&9) ; then
     fd=9
 fi >/dev/null 2>&1
 
-valgrind=""
-if command -v valgrind && test -n "${VALGRIND}" ; then
-    valgrind="valgrind --quiet --log-file=valgrind.log"
-fi >/dev/null 2>&1
+PATH="${PWD}:$PATH"
+
+files_path="${PWD}/test/TestingFiles"
+
+status=0
+
+fatal() {
+    local test="${1}"
+    local message="${2}"
+
+    echo "NOK: ${test}, ${message}" >&${fd}
+    status=1
+    clean
+
+    exit ${status}
+}
+
+clean() {
+     pushd "${files_path}" >/dev/null 2>&1
+        local output="$(git rev-parse --show-cdup)"
+        if [ "${?}" -ne "0" ] || [ -n "${output}" ] ; then
+            echo "NOK: internal, ${files_path} is not a git repository" >&${fd}
+            status=1
+            return ${status}
+        fi
+        git clean -fdx
+    popd >/dev/null 2>&1
+}
+
+contains() {
+    echo "${2}" | grep -q "${1}"
+}
+
+check_success() {
+    local rejected_message="${1}"
+    local accepted_message="${2}"
+    local local_status=0
+
+    if [ "${cmd_status}" -ne "0" ] ; then
+        if [ -n "${cmd_stderr}" ] ; then
+            echo "NOK: ${test}/${file}, ${rejected_message}, ${cmd_stderr}" >&${fd}
+        else
+            echo "NOK: ${test}/${file}, ${rejected_message} without message" >&${fd}
+        fi
+        status=1
+        local_status=1
+    elif contains "Error:" "${cmd_stderr}" ; then
+        echo "NOK: ${test}/${file}, "${accepted_message}" with error message: ${cmd_stderr}" >&${fd}
+        status=1
+        local_status=1
+    fi
+
+    return ${local_status}
+}
+
+check_failure() {
+    local rejected_message="${1}"
+    local accepted_message="${2}"
+    local local_status=0
+
+    if [ "${cmd_status}" -eq "0" ] ; then
+        echo "NOK: ${test}/${file}, ${accepted_message}" >&${fd}
+        status=1
+        local_status=1
+    else
+        if [ ! -n "${cmd_stderr}" ] ; then
+            echo "NOK: ${test}/${file}, ${rejected_message} without message" >&${fd}
+            status=1
+            local_status=1
+        fi
+    fi
+
+    return ${local_status}
+}
+
+check_framemd5() {
+    local raw="${1}"
+    local mkv="${2}"
+    local local_status=0
+
+    local media=$(ffmpeg -hide_banner -i "${raw}" 2>&1 </dev/null | tr -d ' ' | grep -m1 '^Stream#.\+:.\+' | cut -d ':' -f 3)
+    if [ "${media}" == "Video" ] ; then
+        pixfmt=$(ffmpeg -hide_banner -i "${raw}" 2>&1 </dev/null | tr -d ' ' | grep -m1 'Stream#.\+:.\+:Video:.\+,' | cut -d, -f2)
+        ffmpeg -i "${raw}" -f framemd5 "${raw}.framemd5" </dev/null >/dev/null 2>&1 >/dev/null 2>&1
```

Comment:
I presume the second `>/dev/null 2>&1` is not needed.

Reply:
Fixed, thank!